modules/geo-layers/src/tile-layer/tile-layer.js

```diff
@@ -1,135 +1,21 @@
-import {log} from '@deck.gl/core';
 import {CompositeLayer} from '@deck.gl/core';
-import {GeoJsonLayer} from '@deck.gl/layers';
-import TileCache from './utils/tile-cache';
-
-const defaultProps = {
-  renderSubLayers: {type: 'function', value: props => new GeoJsonLayer(props), compare: false},
-  getTileData: {type: 'function', value: ({x, y, z}) => Promise.resolve(null), compare: false},
-  // TODO - change to onViewportLoad to align with Tile3DLayer
-  onViewportLoad: {type: 'function', optional: true, value: null, compare: false},
-  // eslint-disable-next-line
-  onTileError: {type: 'function', value: err => console.error(err), compare: false},
-  maxZoom: null,
-  minZoom: 0,
-  maxCacheSize: null
-};
+import {BaseTileLayer} from '@deck.gl/layers';
+import {tile2boundingBox} from './utils/tile-util';
+import {getTileIndices} from './utils/viewport-util';
 
 export default class TileLayer extends CompositeLayer {
```

Comment:
It might be cheaper to extend the `BaseTileLayer` and overwrite the default props.

Reply:
This was a point I went back and forth on.  I landed on trying to match the other layers in `geo-layers` (which all render sublayers) but of course this layer only renders one sublayer while those generally did more. I will change it, then.